qdrant/v1.0.x/distributed_deployment.md

```diff
@@ -343,10 +346,20 @@ Before responding to the client, the peer handling the request dispatches all op
 - reads are using a partial fan-out strategy to optimize latency and availability
 - writes are executed in parallel on all active sharded replicas
 
-In case of write operations, it is possible to control when the server replies to the client using the write concern factor configuration.
+![Embeddings](/docs/concurrent-operations-replicas.png)
 
-The `write_concern_factor` represents the number of replicas that must acknowledge a write operation before responding to the client. It is set to one by default.
+However, in some cases, it is necessary to ensure addition guarantees during the possible hardware instability, mass concurrent updates of same documents, etc.
+
+Qdrant provides a few options to control consistency guarantees:
+
+- `write_concern_factor` - defines the number of replicas that must acknowledge a write operation before responding to the client. Increasing this value will make write operations tollerant to network partitions in the cluster, but will require higher number of replicas to be active to perform write operations.
+- Read `consistency` param, can be used with search and retrieve operations to ensure that the results obtained from all replicas are the same. In this option is used, qdrant will perform read operation on multiple replicas and resolve the result according to selected strategy. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preffered if update operations are frequent and the number of replicas is low.
+- Write `ordering` param, can be used with update and delete operations to ensure that the operations are executed in the same order on all replicas. If this option is used, qdrant will route the operation to the leader replica of the shard and wait for the response before responding to the client. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preffered if read operations are more frequent then update and search performance is critical.
```

Comment:
```suggestion
- Write `ordering` param, can be used with update and delete operations to ensure that the operations are executed in the same order on all replicas. If this option is used, qdrant will route the operation to the leader replica of the shard and wait for the response before responding to the client. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preferred if hte read operations are more frequent than update and if search performance is critical.
```